airflow/hooks/druid_hook.py

```diff
@@ -83,6 +84,9 @@ def construct_ingest_query(
         else:
             num_shards = -1
 
+        query_granularity = query_granularity or "NONE"
+        segment_granularity = segment_granularity or "DAY"
```

Comment:
Why not have those as keyword defaults in the ingest query instead of requiring a new argument? That way you get backward compatibility for free.